crates/oci-tar-builder/src/lib.rs

```diff
@@ -68,13 +73,17 @@ impl Builder {
         };
 
         for layer in self.layers.iter() {
-            let dgst = try_digest(layer.as_path()).context("failed to digest layer")?;
-            let meta = metadata(layer).context("could not get layer metadata")?;
+            let dgst = try_digest(layer.0.as_path()).context("failed to digest layer")?;
+            let meta = metadata(layer.0.clone()).context("could not get layer metadata")?;
             let oci_digest = "sha256:".to_owned() + &dgst;
 
+            let mut media_type = MediaType::ImageLayer;
+            if !layer.1.is_empty() {
+                media_type = MediaType::Other(layer.1.clone());
```

Comment:
Should the `layer.1` media types be enums instead of strings? Or is the idea that we are just a postman for the types?

Reply:
since the various types for wasm aren't set in stone and the library might be used for other reasons, I was just passing those values through. Maybe in the future we might want to have stricter checks here?